splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSet.java

```diff
@@ -117,6 +117,22 @@ public int partitions() {
         return Pair.newPair(new ControlDataSet(rows.iterator()), rows.size());
     }
 
+    @Override
+    public Pair<DataSet, Integer> persistIt() {
+        return materialize();
+    }
+
+    @Override
+    public void unpersistIt() {
+        return;
+    }
+
+    @Override
+    public Pair<Pair<DataSet, DataSet>, Integer> persistIt2() {
```

Comment:
I don't like the "persistIt2" method too much. Would it be possible to add a getClone() method to the API instead? for SparkDataSet it would return `this', for Control you could return a special subclass in materialize() ( for instance MaterializedControlDataSet extends ControlDataSet ) that contains a List of values and calls super(list.iterator()) on the constructor, the getClone() in this case would return new MaterializedControlDataSet(list);

Reply:
Thanks @dgomezferro, I've made the change accordingly.